projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 1996-2020 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <math.h>
+#include <webots/Keyboard.hpp>
+#include <webots/Motor.hpp>
+#include <webots/Robot.hpp>
+
+using namespace webots;
+using namespace std;
+
+Motor *motor[3];
+
+// move forward vx m/s, left vy m/s, rotate left va radians/s (va=Angular
+// Velocity)
+void setSpeeds(double vx, double vy, double va) {
+  vx /= 0.063;  // wheel radius = 0.063
+  vy /= 0.063;
+  va *= 0.1826 / 0.063;  // distance wheel center to robot center 0.1826
+  motor[0]->setVelocity(vy + va);
+  motor[1]->setVelocity(-0.5 * vy - sqrt(0.75) * vx + va);
+  motor[2]->setVelocity(-0.5 * vy + sqrt(0.75) * vx + va);
+}
```

Comment:
It's generally better to use explicit symbol names (variables, functions) rather than having to write comments to explain them.

```suggestion
static void moveForward(double vx, double vy, double angularVelocity) {
  const double wheelRadius = 0.063;
  vx /= wheelRadius;
  vy /= wheelRadius;
  angularVelocity *= 0.1826 / wheelRadius;  // distance from wheel center to robot center is 0.1826 m.
  motor[0]->setVelocity(vy + angularVelocity);
  motor[1]->setVelocity(-0.5 * vy - sqrt(0.75) * vx + angularVelocity);
  motor[2]->setVelocity(-0.5 * vy + sqrt(0.75) * vx + angularVelocity);
}
```